swarm/animator.py

```
@@ -45,3 +59,306 @@ def animate(i):
+    feedback to the developer.
+    This isn't needed by end-user, but could be used ahead of time if they like.
+
+    Raises:
+        ValueError: in the case of invalid arguments
+    """
+    cls = plt.Axes
+    missing = []
+    for k in kwargs:
+        if not hasattr(cls, k):
+            missing.append(k)
+    if missing:
+        raise ValueError("plt.Axes does not have {}".format(",".join(missing)))
+
+
+def _kwargs_hook(**kwargs):
+    """
+    Our animation library uses hooks to allow once of configuration of an axes. These
+    include things like plotting the true y value or setting sensible x/y limits for the plot
+
+    This can be time consuming to do, so this provides an easy way to do so, allowing
+    one to call the various functions of ``plt.Axes`` as below. This is expected to be fit
+    into the existing machinery of the SwarmPlot sub/duck classes so you won't need to make
+    this call directly
+
+    Examples:
+        >>> _kwargs_hook(set_title="Goofy Experiment", set_ylabel="loss")
+        # will be used mostly with constructors like
+        >>> LineSwarm.standard(x, y, data, set_title="Goofy Experiment", set_ylabel="loss")
+
+    Raises:
+        ValueError: will attempt to validate the kwargs ahead of time, but also on execution
+        in case the function arg is incorrect
+
+    """
+    # get an early error if this is a problem
+    # rather than at animate time
+    _validate_ax_kwargs(**kwargs)
+
+    def inner_hook(ax):
+        for k, v in kwargs.items():
+            try:
+                func = getattr(ax, k)
+            except AttributeError:
+                raise ValueError(f"Used a method {k}")
+            else:
+                try:
+                    func(v)
+                except Exception as e:
+                    # we might have an issue that we're hitting a property
+                    # or have type mismatches
+                    raise ValueError("Unexpected failure") from e
```

So this is just checking for an error?  I do get the desire to check for errors before you animate.  However, in the workflow I did in R I would always just plot a single frame of the animation, since this was easy to do.  Is it not possible here? This would reveal errors, and much more about how things generally looked etc.  I kinda am left wondering if this is a sufficient replacement for such a process, and if it isn't (and one would plot some frames first anyway in the process of preparing something) whether it's necessary? Seems artisanal in the extreme.

---

That's a really good point - I can just call the plot init function instead and that will try and create the first frame. This was meant to help fail early with good information, but I think a dry run might be simpler instead.